spiffworkflow-backend/src/spiffworkflow_backend/services/authentication_service.py

```diff
@@ -87,6 +88,26 @@ class AuthenticationOptionNotFoundError(Exception):
     pass
 
 
+class PKCE:
+    '''
+    Config to support PKCE (Proof Key for Code Exchange)
+    RFC: https://www.rfc-editor.org/rfc/rfc7636
+    '''
+
+    CODE_VERIFIER_KEY = "code_verifier"
+    CODE_CHALLENGE_KEY = "code_challenge"
+    CODE_CHALLENGE_METHOD_KEY = "code_challenge_method"
```

Comment:
The PKCE flow: the `code_challenge_method` (fixed to `"S256"` in this PR for security reasons) and `code_challenge`--a hashed version of `code_verifier`--are sent to the auth server on getting an authorization code.  The `code_verifier` is sent to the auth server during the request that exchanges this authorization code for an access token. This ensures that the client that received the authorization code is the same client requesting the access token.

Reply:
I've scoped everything to this class mostly for organizational reasons.